src/palace/manager/api/admin/controller/report.py

```diff
@@ -22,6 +22,28 @@
 from palace.manager.util.problem_detail import ProblemDetail, ProblemDetailException
 
 
+def _authorize_from_request(
+    request: Request,
+) -> tuple[Admin, Library, None] | tuple[None, None, Response]:
+    """Authorize the admin for the library from the request.
+
+    :param request: A Flask Request object.
+    :return: A 3-tuple of admin, library, and error response.
+        If there is no library, the error response is a 404.
+        Otherwise, we return a valid admin and library, if the admin is authorized.
+    :raise: ProblemDetailException, jf admin not authorized for library.
+    """
+    library: Library | None = getattr(request, "library")
+    if library is None:
+        return None, None, Response(status=404)
```

Comment:
I wonder if instead of creating another problem detail, maybe you can use `.detailed` on `LIBRARY_NOT_FOUND` to updated whatever doesn't fit here.

Reply:
Did a little more digging and decided to go with an unmodified `LIBRARY_NOT_FOUND`, since that's what `BaseCirculationManagerController.library_for_request` [would return](https://github.com/ThePalaceProject/circulation/blob/b1ab61ac9a9506d4ccad11efaeb4df5e3998a84b/src/palace/manager/api/controller/base.py#L125-L126) (via the `@has_library` decorator) anyway.

If I hadn't done that, I probably would've created a new PD, since I would've wanted it in both the api and the tests.